File: 4-1.php
<?php

$input = 'bgvyzdsv';

$number = 0;

$hash = '';

do {
    $number += 1;
    $hash = md5($input . $number);
} while (substr($hash, 0, 5) !== '00000');

var_dump([$number, $hash]);

File: 4/src/bin/4-1.rs
extern crate crypto;

use crypto::digest::Digest;
use crypto::md5::Md5;

fn main() {
    let input = "bgvyzdsv";
    let target = "00000";

    let mut number = 0;

    let mut md5 = Md5::new();

    md5.input_str(&*(format!("{}{}", input, number)));

    while !md5.result_str().starts_with(target) {
        md5.reset();
        number += 1;
        md5.input_str(&*(format!("{}{}", input, number)));
    }

    println!("{}", number);
}
